### ISSUE TITLE
Trigger promise tracker from axios interceptor

### ISSUE THREAD

**Issue by sushmitg** [Lemoncode/react-promise-tracker#75]
Any way we can use this library in axios interceptor file so that I don't have to wrap every axios API call instance in every component?

**Comment by brauliodiez** [comment]
Hello, checking Axios interceptors the are executed before the promise is created or right after the request is completed, there seems no way to handle there the promise.

If you want to track all request by default you could:

a. Build a wrapper around Axios methods (get, post, put...) with the exact same parameters as Axios have then that methods make the call to e.g. trackPromise(get(...)) by following this approach you have a single entry point where you introduce the trackPromise.

b. You can isolated axios calls in an api  file (here you will repeat trackpromise on every call but you have your api's isolated from e.g. component code).

Tracking all promises by default may not be a good idea, you can have the following scenarios:

- Some request you want them to be tracked, some others not (e.g. load data in background),.
- In some cases you may want to block an specific area of the screen.
- Even for the same endpoint you may want different behaviors (when the user first loads the page show loading indicator once we have data in memory, just refresh data silently).

**Comment by brauliodiez** [comment]
We will provide some manual entry points to tweak / reset promise progress counter.

Closing this issue in favor of #76

**Comment by joaoromeira** [comment]
I resolved like this
```javascript
import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_URL,
  headers: {
    Accept: 'application/json',
  },
});

const get = param => trackPromise(api.get(param));

const post = (param, body) => trackPromise(api.post(param, body));

const put = (param, body) => trackPromise(api.put(param, body));

const del = param => trackPromise(api.delete(param));

export default {
  get,
  post,
  put,
  delelete: del,
};
```

**Comment by brauliodiez** [comment]
Nice approach @Vini-Dev , we have released as well manual endpoints, check #76 

manuallyIncrementPromiseCounter
manuallyIncrementPromiseCounter

Use with "care" :), more info in our documentation: https://lemoncode.github.io/react-promise-tracker/docs/api.html#manuallyincrementpromisecounter
